src/main/java/me/ryanhamshire/GPFlags/util/ChatUtil.java

```diff
@@ -0,0 +1,258 @@
+        }
+        return message;
+    }
+
+    public static Component fixColor(String message, String key, String value) {
+        if (message.split(key).length == 0)
+            return ChatUtil.hexComp(value);
+
+        String colorMsg = message.split(key)[0];
+        String color = colorMsg.substring(colorMsg.length() - 2);
+        if (color.trim().isEmpty())
+            return ChatUtil.hexComp(value);
+        if (color.charAt(0) == '&')
+            return ChatUtil.hexComp(color + value);
+        else return ChatUtil.hexComp(value);
+    }
+
+    private static String getNativeColor(char color) {
+        String reset = "<!b><!i><!obf><!u><!st>";
+        String colorCode = "<color:white>";
+        if (color == 'r') return reset + colorCode;
+        switch (color) {
+            case 'a':
+                colorCode = "<color:green>";
+            case 'b':
+                colorCode = "<color:aqua>";
+            case 'c':
+                colorCode = "<color:red>";
+            case 'd':
+                colorCode = "<color:light_purple>";
+            case 'e':
+                colorCode = "<color:yellow>";
+            case 'f':
+                colorCode = "<color:white>";
+            case '0':
+                colorCode = "<color:black>";
+            case '1':
+                colorCode = "<color:dark_blue>";
+            case '2':
+                colorCode = "<color:dark_green>";
+            case '3':
+                colorCode = "<color:dark_aqua>";
+            case '4':
+                colorCode = "<color:dark_red>";
+            case '5':
+                colorCode = "<color:dark_purple>";
+            case '6':
+                colorCode = "<color:gold>";
+            case '7':
+                colorCode = "<color:gray>";
+            case '8':
+                colorCode = "<color:dark_gray>";
+            case '9':
+                colorCode = "<color:blue>";
+            case 'l':
+                colorCode = "<bold>";
+            case 'm':
+                colorCode = "<strikethrough>";
+            case 'n':
+                colorCode = "<u>";
+            case 'o':
+                colorCode = "<italic>";
+            case 'k':
+                colorCode = "<obf>";
+        }
+        switch (color) {
+            case 'l':
+            case 'm':
+            case 'n':
+            case 'o':
+            case 'k':
+                break;
+            default:
+                colorCode = reset + colorCode;
+        }
+        return colorCode;
+    }
+
+    private static String validateMinimessage(CommandSender sender, String message) {
+
+        // Check if there is a link in the message without click-component
+        Pattern pattern = Pattern.compile("https?://\\S+");
+        Matcher matcher = pattern.matcher(message);
+        while (matcher.find()) {
+            boolean requiredChanges = false;
+            if(!sender.hasPermission("gpflags.messages.links")) {
```

Comment:
Same problem with people that have the permission, I think it would loop until no memory? 
My proposal:
- Just parse the legacy link always into click-component `open_url` and keep checking for `gpflags.messages.click.open_url` instead
- This way we dont have inconsistency in two permission nodes for the same use-case

Reply:
I updated the branch with the approach and a previous check if it already is a click-component